File: 2543-1941-649-dota2-senate/2543-1941-649-dota2-senate.js
/**
 * @param {string} senate
 * @return {string}
 */
var predictPartyVictory = function(senate) {
    const R = [],D = []

    for(let i=0;i<senate.length;i++){
        if(senate[i]==='R')
            R.push(i)
        else
            D.push(i)
    }

    while(R.length>0 && D.length>0){
        let r = R.shift(),d = D.shift()
        if(r<d)
            R.push(senate.length+r)
        else
            D.push(senate.length+d)
    }
    if(R.length===0)
        return 'Dire'
    return 'Radiant'
};

File: 2543-1941-649-dota2-senate/2543-1941-649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        R,D = deque(),deque()
        for i in range(len(senate)):
            if senate[i]=='R':
                R.append(i)
            else:
                D.append(i)

        while len(R)>0 and len(D)>0:
            r,d = R.popleft(),D.popleft()
            if r<d:
                R.append(len(senate)+r)
            else:
                D.append(len(senate)+d)
        
        if len(R)==0:
            return 'Dire'
        return 'Radiant'